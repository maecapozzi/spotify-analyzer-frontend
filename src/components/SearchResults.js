import React from 'react'
import SongLink from './SongLink'

const SearchResults = props => (
  <div>
    <ul>
      {props.results.map((result) => {
        return <SongLink key={result.id} id={result.id} name={result.name} artist={result.artist} />
      })}
    </ul>
  </div>
)

export default SearchResults
