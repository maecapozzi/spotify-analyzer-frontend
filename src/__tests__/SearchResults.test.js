import React from 'react'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

import SearchResults from '../components/SearchResults'

configure({ adapter: new Adapter() })

test('it renders correctly', () => {
  const searchResultsProps = {
    handleClick: () => {
      console.log('needs logic')
    },
    results: [
      {
        artist: 'Luis Fonsi',
        id: '6rPO02ozF3bM7NnOV4h6s2',
        name: 'Despacito - Remix'
      },
      {
        artist: 'Luis Fonsi',
        id: '7CUYHcu0RnbOnMz4RuN07w',
        name: 'Despacito (Featuring Daddy Yankee)'
      },
      {
        artist: 'Madilyn Bailey',
        id: '4vxA3aI7l73i0Hi819OQhH',
        name: 'Despacito'
      }
    ]
  }

  const component = shallow(<SearchResults {...searchResultsProps} />)
  expect(component).toMatchSnapshot()
})
