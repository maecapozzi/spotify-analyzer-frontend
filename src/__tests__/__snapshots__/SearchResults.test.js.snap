// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    handleClick={[Function]}
    results={
        Array [
            Object {
              "artist": "Luis Fonsi",
              "id": "6rPO02ozF3bM7NnOV4h6s2",
              "name": "Despacito - Remix",
            },
            Object {
              "artist": "Luis Fonsi",
              "id": "7CUYHcu0RnbOnMz4RuN07w",
              "name": "Despacito (Featuring Daddy Yankee)",
            },
            Object {
              "artist": "Madilyn Bailey",
              "id": "4vxA3aI7l73i0Hi819OQhH",
              "name": "Despacito",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Paper
        circle={false}
        rounded={true}
        style={
                Object {
                        "backgroundColor": "#181A1B",
                        "margin": "0 auto",
                        "maxWidth": 800,
                        "padding": "20px",
                      }
        }
        transitionEnabled={true}
        zDepth={1}
>
        <div
                className="search-results--p"
        >
                <SongLink
                        handleClick={[Function]}
                        result={
                                Object {
                                        "artist": "Luis Fonsi",
                                        "id": "6rPO02ozF3bM7NnOV4h6s2",
                                        "name": "Despacito - Remix",
                                      }
                        }
                />
                <SongLink
                        handleClick={[Function]}
                        result={
                                Object {
                                        "artist": "Luis Fonsi",
                                        "id": "7CUYHcu0RnbOnMz4RuN07w",
                                        "name": "Despacito (Featuring Daddy Yankee)",
                                      }
                        }
                />
                <SongLink
                        handleClick={[Function]}
                        result={
                                Object {
                                        "artist": "Madilyn Bailey",
                                        "id": "4vxA3aI7l73i0Hi819OQhH",
                                        "name": "Despacito",
                                      }
                        }
                />
        </div>
</Paper>,
      "className": "search-results__container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="search-results--p"
>
          <SongLink
                    handleClick={[Function]}
                    result={
                              Object {
                                        "artist": "Luis Fonsi",
                                        "id": "6rPO02ozF3bM7NnOV4h6s2",
                                        "name": "Despacito - Remix",
                                      }
                    }
          />
          <SongLink
                    handleClick={[Function]}
                    result={
                              Object {
                                        "artist": "Luis Fonsi",
                                        "id": "7CUYHcu0RnbOnMz4RuN07w",
                                        "name": "Despacito (Featuring Daddy Yankee)",
                                      }
                    }
          />
          <SongLink
                    handleClick={[Function]}
                    result={
                              Object {
                                        "artist": "Madilyn Bailey",
                                        "id": "4vxA3aI7l73i0Hi819OQhH",
                                        "name": "Despacito",
                                      }
                    }
          />
</div>,
        "circle": false,
        "rounded": true,
        "style": Object {
          "backgroundColor": "#181A1B",
          "margin": "0 auto",
          "maxWidth": 800,
          "padding": "20px",
        },
        "transitionEnabled": true,
        "zDepth": 1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SongLink
              handleClick={[Function]}
              result={
                            Object {
                                          "artist": "Luis Fonsi",
                                          "id": "6rPO02ozF3bM7NnOV4h6s2",
                                          "name": "Despacito - Remix",
                                        }
              }
/>,
            <SongLink
              handleClick={[Function]}
              result={
                            Object {
                                          "artist": "Luis Fonsi",
                                          "id": "7CUYHcu0RnbOnMz4RuN07w",
                                          "name": "Despacito (Featuring Daddy Yankee)",
                                        }
              }
/>,
            <SongLink
              handleClick={[Function]}
              result={
                            Object {
                                          "artist": "Madilyn Bailey",
                                          "id": "4vxA3aI7l73i0Hi819OQhH",
                                          "name": "Despacito",
                                        }
              }
/>,
          ],
          "className": "search-results--p",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "6rPO02ozF3bM7NnOV4h6s2",
            "nodeType": "function",
            "props": Object {
              "handleClick": [Function],
              "result": Object {
                "artist": "Luis Fonsi",
                "id": "6rPO02ozF3bM7NnOV4h6s2",
                "name": "Despacito - Remix",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "7CUYHcu0RnbOnMz4RuN07w",
            "nodeType": "function",
            "props": Object {
              "handleClick": [Function],
              "result": Object {
                "artist": "Luis Fonsi",
                "id": "7CUYHcu0RnbOnMz4RuN07w",
                "name": "Despacito (Featuring Daddy Yankee)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4vxA3aI7l73i0Hi819OQhH",
            "nodeType": "function",
            "props": Object {
              "handleClick": [Function],
              "result": Object {
                "artist": "Madilyn Bailey",
                "id": "4vxA3aI7l73i0Hi819OQhH",
                "name": "Despacito",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Paper
          circle={false}
          rounded={true}
          style={
                    Object {
                              "backgroundColor": "#181A1B",
                              "margin": "0 auto",
                              "maxWidth": 800,
                              "padding": "20px",
                            }
          }
          transitionEnabled={true}
          zDepth={1}
>
          <div
                    className="search-results--p"
          >
                    <SongLink
                              handleClick={[Function]}
                              result={
                                        Object {
                                                  "artist": "Luis Fonsi",
                                                  "id": "6rPO02ozF3bM7NnOV4h6s2",
                                                  "name": "Despacito - Remix",
                                                }
                              }
                    />
                    <SongLink
                              handleClick={[Function]}
                              result={
                                        Object {
                                                  "artist": "Luis Fonsi",
                                                  "id": "7CUYHcu0RnbOnMz4RuN07w",
                                                  "name": "Despacito (Featuring Daddy Yankee)",
                                                }
                              }
                    />
                    <SongLink
                              handleClick={[Function]}
                              result={
                                        Object {
                                                  "artist": "Madilyn Bailey",
                                                  "id": "4vxA3aI7l73i0Hi819OQhH",
                                                  "name": "Despacito",
                                                }
                              }
                    />
          </div>
</Paper>,
        "className": "search-results__container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="search-results--p"
>
            <SongLink
                        handleClick={[Function]}
                        result={
                                    Object {
                                                "artist": "Luis Fonsi",
                                                "id": "6rPO02ozF3bM7NnOV4h6s2",
                                                "name": "Despacito - Remix",
                                              }
                        }
            />
            <SongLink
                        handleClick={[Function]}
                        result={
                                    Object {
                                                "artist": "Luis Fonsi",
                                                "id": "7CUYHcu0RnbOnMz4RuN07w",
                                                "name": "Despacito (Featuring Daddy Yankee)",
                                              }
                        }
            />
            <SongLink
                        handleClick={[Function]}
                        result={
                                    Object {
                                                "artist": "Madilyn Bailey",
                                                "id": "4vxA3aI7l73i0Hi819OQhH",
                                                "name": "Despacito",
                                              }
                        }
            />
</div>,
          "circle": false,
          "rounded": true,
          "style": Object {
            "backgroundColor": "#181A1B",
            "margin": "0 auto",
            "maxWidth": 800,
            "padding": "20px",
          },
          "transitionEnabled": true,
          "zDepth": 1,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <SongLink
                handleClick={[Function]}
                result={
                                Object {
                                                "artist": "Luis Fonsi",
                                                "id": "6rPO02ozF3bM7NnOV4h6s2",
                                                "name": "Despacito - Remix",
                                              }
                }
/>,
              <SongLink
                handleClick={[Function]}
                result={
                                Object {
                                                "artist": "Luis Fonsi",
                                                "id": "7CUYHcu0RnbOnMz4RuN07w",
                                                "name": "Despacito (Featuring Daddy Yankee)",
                                              }
                }
/>,
              <SongLink
                handleClick={[Function]}
                result={
                                Object {
                                                "artist": "Madilyn Bailey",
                                                "id": "4vxA3aI7l73i0Hi819OQhH",
                                                "name": "Despacito",
                                              }
                }
/>,
            ],
            "className": "search-results--p",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "6rPO02ozF3bM7NnOV4h6s2",
              "nodeType": "function",
              "props": Object {
                "handleClick": [Function],
                "result": Object {
                  "artist": "Luis Fonsi",
                  "id": "6rPO02ozF3bM7NnOV4h6s2",
                  "name": "Despacito - Remix",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "7CUYHcu0RnbOnMz4RuN07w",
              "nodeType": "function",
              "props": Object {
                "handleClick": [Function],
                "result": Object {
                  "artist": "Luis Fonsi",
                  "id": "7CUYHcu0RnbOnMz4RuN07w",
                  "name": "Despacito (Featuring Daddy Yankee)",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "4vxA3aI7l73i0Hi819OQhH",
              "nodeType": "function",
              "props": Object {
                "handleClick": [Function],
                "result": Object {
                  "artist": "Madilyn Bailey",
                  "id": "4vxA3aI7l73i0Hi819OQhH",
                  "name": "Despacito",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
